crates/cadenza-gcode/src/gcode_syntax.rs

```diff
@@ -0,0 +1,263 @@
+            errors: Vec::new(),
+        }
+    }
+
+    fn parse(mut self) -> Parse {
+        self.builder.start_node(Kind::Root.into());
+
+        for line in self.src.lines() {
+            let line_start = self.offset;
+            self.parse_line(line, line_start);
+            self.offset = line_start + line.len() + 1; // +1 for newline
+        }
+
+        self.builder.finish_node();
+
+        Parse {
+            green: self.builder.finish(),
+            errors: self.errors,
+        }
+    }
+
+    fn parse_line(&mut self, line: &str, line_start: usize) {
+        let line = line.trim();
+
+        if line.is_empty() {
+            return;
+        }
+
+        // Handle comments
+        if line.starts_with(';') {
+            // Comments are trivia, we can skip them or add as whitespace
+            return;
+        }
+
+        // Handle inline comments
+        let code_part = if let Some(pos) = line.find(';') {
+            line[..pos].trim()
+        } else {
+            line
+        };
+
+        if code_part.is_empty() {
+            return;
+        }
+
+        // Parse command as an Apply node
+        self.parse_command(code_part, line_start);
+    }
+
+    fn parse_command(&mut self, code: &str, line_start: usize) {
+        let parts: Vec<&str> = code.split_whitespace().collect();
+        if parts.is_empty() {
+            return;
+        }
+
+        // Start an Apply node for the command
+        self.builder.start_node(Kind::Apply.into());
+
+        // Parse command code (G1, M104, etc.) as the receiver
+        self.builder.start_node(Kind::ApplyReceiver.into());
+        self.parse_command_identifier(parts[0], line_start);
+        self.builder.finish_node();
+
+        // Parse parameters as key-value pair arguments: [=, key, value]
+        for param in &parts[1..] {
+            self.builder.start_node(Kind::ApplyArgument.into());
+            self.parse_parameter_as_assignment(param, line_start);
+            self.builder.finish_node();
+        }
+
+        self.builder.finish_node();
+    }
+
+    fn parse_command_identifier(&mut self, cmd: &str, _line_start: usize) {
+        // Create an Identifier node (not just a token)
+        self.builder.start_node(Kind::Identifier.into());
+        self.builder.token(Kind::Identifier.into(), cmd);
+        self.builder.finish_node();
+    }
+
+    fn parse_parameter_as_assignment(&mut self, param: &str, _line_start: usize) {
+        if param.is_empty() {
+            return;
+        }
+
+        let letter = param.chars().next().unwrap();
+        let value_str = &param[1..];
+
+        // Create an Apply node for assignment: [=, key, value]
+        self.builder.start_node(Kind::Apply.into());
+
+        // The "=" operator as receiver
+        self.builder.start_node(Kind::ApplyReceiver.into());
+        self.builder.start_node(Kind::Identifier.into());
+        self.builder.token(Kind::Identifier.into(), "=");
```

Comment:
So the problem with this is it's creating a token out of thin air, which is going to mess up the offsets in the tree. I wonder if we keep it simple and just have the letter be the receiver instead. Otherwise we'd have to add a synthetic `=` token and update the eval in a bunch of places, which sounds more painful